### PR TITLE
Custom hook for draft saving

### LINKED ISSUE #129: Custom hook for draft saving
#### Proposed solution

Draft saving logic is written in both `WizardFillObjectDetailsForm` and `WizardAlert` components. This logic should be only in one place and reusable. Custom hook could be answer for this.

#### DoD (Definition of Done)

Draft saving logic should be in one place and reusable.

### PR DESCRIPTION
**Pull request by saulipurhonen**
### Description

Custom hook for reducing duplicate code for draft saving logic in both `WizardFillObjectDetailsForm` and `WizardAlert` components.

### Related issues

Closes #129 

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Changes Made

- Separate draft save logic into dedicated hook
- Renamed `currentObject.type` property to `currentObject.status` since this name more corresponding
- Display `Update a draft` instead of `Save as Draft` in form action button when editing a draft.

### Testing

- [x] Tests do not apply

### CODE REVIEW

**Review by blankdots — APPROVED** [review]
this looks like referactoring, and LGTM

File: src/components/NewDraftWizard/WizardForms/WizardFillObjectDetailsForm.js [hunk 1]
```diff
@@ -156,17 +157,17 @@ const CustomCardHeader = (props: CustomCardHeaderProps) => {
         Clear form
       </Button>
       <Button variant="contained" aria-label="save form as draft" size="small" onClick={onClickSaveDraft}>
-        Save as Draft
+        {currentObject?.status === "Draft" ? "Update draft" : " Save as Draft"}
```
Comment (hannyle) [inline src/components/NewDraftWizard/WizardForms/WizardFillObjectDetailsForm.js:160]:
```suggestion
{currentObject?.status === ObjectStatus.draft ? "Update draft" : " Save as Draft"}
```

Comment (hannyle) [reply src/components/NewDraftWizard/WizardForms/WizardFillObjectDetailsForm.js:160]: just this small thing

**Review by hannyle — APPROVED** [review]
looks good to me.

**Review by lilachic — APPROVED** [review]
Looks good to me.

**Pull request merged**